MonoGame.Framework/Media/VideoPlayer.WMS.cs

```diff
@@ -59,102 +87,132 @@ private void PlatformInitialize()
             MediaFactory.CreateMediaSession(null, out _session);
         }
 
-        private Texture2D PlatformGetTexture()
+        private void CreateTexture()
         {
-            var sampleGrabber = _currentVideo.SampleGrabber;
+            if (_currentVideo == null)
+                return;
 
-            var texData = sampleGrabber.TextureData;
+            if (_texture != null)
+            {
+                // If the new video is different in size to the previous video, dispose of the current texture
+                if (_texture.Width != _currentVideo.Width || _texture.Height != _currentVideo.Height)
+                {
+                    _texture.Dispose();
+                    _texture = null;
+                }
+            }
 
-            if (texData == null)
-                return null;
+            if (_texture == null)
+                _texture = new Texture2D(Game.Instance.GraphicsDevice, _currentVideo.Width, _currentVideo.Height, false, SurfaceFormat.Bgr32);
```

Comment:
It would be nice to get these Game.Instance references out of the code at least for desktop. That makes it easier for people to embed MG (#4730) or write a custom Game class and potentially we can allow running multiple games at once. 

For this class it would be nice if it could be run without a Game instance being present too. I can think of two ways to do this, one that breaks with the XNA API and one that doesn't (but it's ugly).

1. Have VideoPlayer take a GraphicsDevice in its construction and store it. This obviously would break XNA games using the VideoPlayer, but it's just a small change and people will notice at compile time. I actually prefer this because IMO it's cleaner than option 2 and it makes sense to have a video player linked to a GraphicsDevice this way.
2. Have some global static service container that we can pull a GraphicsDevice or a GraphicsDeviceManager (and throught that a GraphicsDevice) from. GD or GDM would register on construction just like how GDM does that for Game currently. From a software design point of view I think this is ewwww.